release.md

```diff
@@ -82,6 +82,16 @@ After the release pull request has been merged into the `main` branch, it is tim
 
 ## Release on conda-forge
 
+In order to release on conda-forge, you can either wait for a bot to create a PR, or manually kickoff the creation with GitHub Actions
```

Comment:
After we've confirmed the release workflow can kick off the action we should update this section

Reply:
Agreed.